### ISSUE TITLE
Not running at all. SignalR error: Undefained

### ISSUE THREAD

**Issue by dabor238** [olofd/react-native-signalr#14]
I try to use this package, but the following error appears. Any help would be really appreciated.

My server info is:
var connection = signalr.hubConnection('https://www.multidoctores.com');
var proxy = connection.createHubProxy('chatHub');

This info is already tasted in Web, Android and Ios native apps.

<img width="703" alt="screen shot 2017-01-10 at 3 33 26 pm" src="https://cloud.githubusercontent.com/assets/1827355/21823681/381d7ecc-d74a-11e6-8cc7-3ecbf220f1ee.png">

<img width="683" alt="screen shot 2017-01-10 at 3 34 26 pm" src="https://cloud.githubusercontent.com/assets/1827355/21823708/55c54734-d74a-11e6-8846-6dd55d64e782.png">

**Comment by olofd** [comment]
Anything related to #12?

**Comment by dabor238** [comment]
I don't think that is the same issue, because:
1) the server is HTTPS 
2) the hub has no security, you can access it here: https://www.multidoctores.com/signalr/hubs

**Comment by dabor238** [comment]
Do I have to configure the hub in any specific way to test the library. Do you have an example of a server side hub that actually works with this library?

**Comment by dabor238** [comment]
It tries to negotiate with

[16:41:33 GMT-0500 (ECT)] SignalR: Negotiating with 'https://www.multidoctores.com/signalr/negotiate?clientProtocol=1.5&connectionData=%5B%7B%22name%22%3A%22chathub%22%7D%5D'.

that brings a valid token. but then it gives us the
SignalR error: undefined.

**Comment by olofd** [comment]
Testing now. Hold on.

**Comment by olofd** [comment]
@dabor238 I get:
"An SSL error has occurred and a secure connection to the server cannot be made."
Sry. There is a bug in the lib that hides the actual error.

**Comment by olofd** [comment]
But do this:
Start react-native with chrome dev tools.
Go to sources tab and press cmd + O (mac) ctrl + O (windows) and search for the file ajax.js
should look like this:

`
export default function(options, data) {
  var request = new XMLHttpRequest();
  request.onreadystatechange = (e) => {
    if (request.readyState !== 4) {
      return;
    }

    if (request.status === 200 && !request._hasError) {
      options.success && options.success(JSON.parse(request.responseText));
    } else {
      options.error && options.error(request);
    }
  };

  request.open(options.type, options.url);
  request.setRequestHeader('content-type', options.contentType);

  request.send(options.data.data && `data=${options.data.data}`);
  
  return {
    abort: function(reason) {
      return request.abort(reason);
    }
  };
}
` 

Place a breakpoint at options.error and then inspect your error.

**Comment by olofd** [comment]
Now for the error itself I do not know what's wrong exactly. You have any idea?

**Comment by olofd** [comment]
I now connected successfully to your server by disabling `NSAppTransportSecurity` don't really know why it won't connect without it.

added 

~~~~~
  <key>NSAppTransportSecurity</key>
   <dict>

      <key>NSAllowsArbitraryLoads</key>
      <true/>

   </dict>
~~~~~~
to Info.plist.

**Comment by dabor238** [comment]
<img width="1108" alt="screen shot 2017-01-10 at 5 24 42 pm" src="https://cloud.githubusercontent.com/assets/1827355/21827481/cb7eb294-d759-11e6-9944-52d5926cbd1f.png">

_response: An SSL error has occurred and a secure connection to the server cannot be made.

No idea why this happened, we are using this hub on WEB, Native Android and Native Ios, and it works fine.

**Comment by dabor238** [comment]
<img width="255" alt="screen shot 2017-01-10 at 5 30 49 pm" src="https://cloud.githubusercontent.com/assets/1827355/21827644/97df1824-d75a-11e6-8537-fd4b144b35c1.png">

I added at the end, my final info.ptlist looks like, but still have the same error. Thanks very much for your help. Do you know what am I doing wrong?

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>en</string>
	<key>CFBundleExecutable</key>
	<string>$(EXECUTABLE_NAME)</string>
	<key>CFBundleIdentifier</key>
	<string>org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>$(PRODUCT_NAME)</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleSignature</key>
	<string>????</string>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>LSRequiresIPhoneOS</key>
	<true/>
	<key>UILaunchStoryboardName</key>
	<string>LaunchScreen</string>
	<key>UIRequiredDeviceCapabilities</key>
	<array>
		<string>armv7</string>
	</array>
	<key>UISupportedInterfaceOrientations</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
		<string>UIInterfaceOrientationLandscapeLeft</string>
		<string>UIInterfaceOrientationLandscapeRight</string>
	</array>
	<key>UIViewControllerBasedStatusBarAppearance</key>
	<false/>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string></string>

	
	<dict>
		<key>NSExceptionDomains</key>
		<dict>
			<key>localhost</key>
			<dict>
				<key>NSExceptionAllowsInsecureHTTPLoads</key>
				<true/>
			</dict>
		</dict>
	</dict>
</dict>

<key>NSAppTransportSecurity</key>
   <dict>

      <key>NSAllowsArbitraryLoads</key>
      <true/>

   </dict>

</plist>

**Comment by dabor238** [comment]
I added at the end and still doesnt work.

<img width="255" alt="screen shot 2017-01-10 at 5 30 49 pm" src="https://cloud.githubusercontent.com/assets/1827355/21827700/cd2e6bba-d75a-11e6-97eb-2c9197007436.png">
<img width="542" alt="screen shot 2017-01-10 at 5 32 56 pm" src="https://cloud.githubusercontent.com/assets/1827355/21827712/d8d57526-d75a-11e6-9ee2-d0f8405d1284.png">

**Comment by olofd** [comment]
Hmm.. might be incorrectly formated.

This i my entire Info.plist in XML:

~~~~~~
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>CFBundleDevelopmentRegion</key>
    <string>en</string>
    <key>CFBundleExecutable</key>
    <string>$(EXECUTABLE_NAME)</string>
    <key>CFBundleIdentifier</key>
    <string>org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)</string>
    <key>CFBundleInfoDictionaryVersion</key>
    <string>6.0</string>
    <key>CFBundleName</key>
    <string>$(PRODUCT_NAME)</string>
    <key>CFBundlePackageType</key>
    <string>APPL</string>
    <key>CFBundleShortVersionString</key>
    <string>1.0</string>
    <key>CFBundleSignature</key>
    <string>????</string>
    <key>CFBundleVersion</key>
    <string>1</string>
    <key>LSRequiresIPhoneOS</key>
    <true/>
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>
    <key>UIRequiredDeviceCapabilities</key>
    <array>
      <string>armv7</string>
    </array>
    <key>UISupportedInterfaceOrientations</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
      <string>UIInterfaceOrientationLandscapeLeft</string>
      <string>UIInterfaceOrientationLandscapeRight</string>
    </array>
    <key>UIViewControllerBasedStatusBarAppearance</key>
    <false/>
    <key>NSLocationWhenInUseUsageDescription</key>
    <string/>
    <key>NSAppTransportSecurity</key>
   <dict>
      <key>NSAllowsArbitraryLoads</key>
      <true/>
   </dict>
  </dict>
</plist>
~~~~~~


Try opening yours in xcode and see if it can open it.

**Comment by dabor238** [comment]
olofd: the boy wonder, the kid, the miracle man. the phenomenon, the unstoppable. It Works.

<img width="1215" alt="screen shot 2017-01-10 at 5 43 36 pm" src="https://cloud.githubusercontent.com/assets/1827355/21828106/bc47f684-d75c-11e6-841f-21ef3fe538e1.png">

**Comment by olofd** [comment]
Haha 👍 Happy to help out. But I will take a look at two things:
1. Why you need that Info.plist edit (will also make a mention about these issues in the readme)
2. That `undefined` error-value needs fixing.

Will do that in the next couple of days when I have time.

But let me know if there is anything else and good luck with your project!

**Comment by dabor238** [comment]
My dear friend.

look at this http://stackoverflow.com/questions/32495059/ios9-will-nsallowsarbitraryloads-true-be-allowed-for-a-release-in-the-app-store

It appears that this fix will couse some problems for a release in the app store.

**Comment by olofd** [comment]
@dabor238 I found the root issue.
Your server uses TLS 1.0. Old and insecure.
So xcode refused it. You should look at upgrading.
But. You what can do this and set an exception for your domain only to use TLS1.0.
This should have no problem in App Store. (Even though the other version with allowing all is no prob. either, have released multiple apps with it.)

Here is the Info.plist with the exception:
~~~~~
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>en</string>
	<key>CFBundleExecutable</key>
	<string>$(EXECUTABLE_NAME)</string>
	<key>CFBundleIdentifier</key>
	<string>org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>$(PRODUCT_NAME)</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleSignature</key>
	<string>????</string>
	<key>CFBundleVersion</key>
	<string>1</string>
	<key>LSRequiresIPhoneOS</key>
	<true/>
	<key>UILaunchStoryboardName</key>
	<string>LaunchScreen</string>
	<key>UIRequiredDeviceCapabilities</key>
	<array>
		<string>armv7</string>
	</array>
	<key>UISupportedInterfaceOrientations</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
		<string>UIInterfaceOrientationLandscapeLeft</string>
		<string>UIInterfaceOrientationLandscapeRight</string>
	</array>
	<key>UIViewControllerBasedStatusBarAppearance</key>
	<false/>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string></string>
	<key>NSAppTransportSecurity</key>
	<dict>
		<key>NSAllowsArbitraryLoads</key>
		<false/>
		<key>NSExceptionDomains</key>
		<dict>
			<key>localhost</key>
			<dict>
				<key>NSTemporaryExceptionAllowsInsecureHTTPLoads</key>
				<true/>
			</dict>
			<key>www.multidoctores.com</key>
			<dict>
				<key>NSTemporaryExceptionMinimumTLSVersion</key>
				<string>TLSv1.0</string>
			</dict>
		</dict>
	</dict>
</dict>
</plist>
~~~~~

**Comment by dabor238** [comment]
Thank so much for the information

**Comment by olofd** [comment]
@dabor238 FYI: I've just released 0.0.11 on npm. I should contain a fix for the `undefined error`-problem you faced earlier. Please update and let me know if something is not working. Have you had any other issues?
